youtube_dl/extractor/common.py

```diff
@@ -354,6 +355,17 @@ def _request_webpage(self, url_or_request, video_id, note=None, errnote=None, fa
                 self.to_screen('%s' % (note,))
             else:
                 self.to_screen('%s: %s' % (video_id, note))
+        if data or headers or query:
+            headers = headers or {}
+            if isinstance(url_or_request, compat_str):
+                if query:
+                    url_or_request = update_url_query(url_or_request, query)
+                url_or_request = sanitized_Request(url_or_request, data, headers)
+            else:
+                url_or_request.full_url = update_url_query(url_or_request.get_full_url(), query)
```

Comment:
`full_url` are available from python 3.1.
`_Request__original` works for python 2(at least version 2.7).
i will try to look at the source code of python 3.0 and see if one of them can work for it.